### ISSUE TITLE
cy.screenshot breaks timers in app

### ISSUE THREAD

**Issue by acthp** [cypress-io/cypress#2432]
### Current behavior:

After ```cy.screenshot```, an infinite sequence of errors is thrown due to corrupt timers.

### Desired behavior:

Timers don't become corrupted.

### Steps to reproduce:

Sorry I don't have a small failing example. The app code looks like this:

```
Rx.Observable.of(true)
    .concat(Rx.Observable.fromEvent(window, 'resize'))
    .debounceTime(200).subscribe(() =>
        callback(['viewportWidth', document.documentElement.clientWidth]));
```

which allows the app to adjust layout after a screen resize. From poking around the ```screenshot``` code, it apparently defers ```setInterval```, ```setTimeout```, and ```requestAnimationFrame``` calls, then resizes the page, screenshots, resizes again, and processes the deferred calls.

I don't know the precise failure mode, but it's the rxjs ```debounceTime``` scheduler that is broken after this call. It will run endlessly, throwing errors due to the action being stale. I suspect the problem is that the cypress timer shims did not consider ```clearInterval``` and friends. So, the debounce mechanism will set an interval, receive a resize event, then receive another resize event, causing it to clear the interval. Clearing the interval does nothing due to the return value from cypress' shimmed method being meaningless (return value of [].push), and then cypress processes its call queue, invoking the original ```setInterval```. So instead of ```setInterval``` followed by ```clearInterval```, it will run ```clearInterval``` on a meaningless value, followed by ```setInterval```. That's just a guess, based on the code & behavior.

I expect I can work around it for now by introducing an out-of-band flag to filter the resize events while screenshotting.

### Versions
cypress 3.1.0, OS ubuntu 14.04, browser chrome 65.0.3325.181

**Comment by brian-mann** [comment]
That is indeed what is happening per here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/cy.coffee#L171

@chrisbreiding we need to collect `clearInterval`, `clearTimeout`, and `cancelAnimationFrame` the same way as the others

**Comment by mikkelking** [comment]
I am hitting a similar problem with an rxjs debounce failing when run under cypress. Do we have an ETA for this fix being released, or is  there a beta I can try?

**Comment by jennifer-shehane** [comment]
This fix was released in Cypress [3.1.1](https://on.cypress.io/changelog#3-1-1)

@mikkelking Please [open a new issue ](https://github.com/cypress-io/cypress/issues/new?template=bug_report.md)detailing your issue including reproducible application code and test code.

**Comment by mikkelking** [comment]
Thanks for the heads up on that, but I'm already on 3.1.4. The problem is with a debounce using rxjs. I worked around it by not using rxjs when it's doing it's CI run with Cypress. I'll submit a new issue so you can have a look at it.
